src/avatar.cpp

```diff
@@ -1654,12 +1654,58 @@ void avatar::add_gained_calories( int cal )
     calorie_diary.front().gained += cal;
 }
 
+void avatar::log_activity_level( float level )
+{
+    calorie_diary.front().activity_levels[level]++;
+}
+
+static const std::map<float, std::string> activity_levels_str = {
+    { NO_EXERCISE, "NO_EXERCISE" },
+    { LIGHT_EXERCISE, "LIGHT_EXERCISE" },
+    { MODERATE_EXERCISE, "MODERATE_EXERCISE" },
+    { ACTIVE_EXERCISE, "ACTIVE_EXERCISE" },
+    { EXTRA_EXERCISE, "EXTRA_EXERCISE" }
+};
+void avatar::daily_calories::save_activity( JsonOut &json ) const
+{
+    json.member( "activity" );
+    json.start_object();
+    for( const std::pair<const float, int> &level : activity_levels ) {
+        json.member( activity_levels_str.at( level.first ), level.second );
+    }
+    json.end_object();
+}
+
+void avatar::daily_calories::read_activity( JsonObject &data )
+{
+    JsonObject jo = data.get_object( "activity" );
+    for( const std::pair<const float, std::string> &member : activity_levels_str ) {
+        int times;
+        jo.read( member.second, times );
+        activity_levels.at( member.first ) = times;
+    }
+}
+
 std::string avatar::total_daily_calories_string() const
 {
     std::string ret = "      gained     spent      total\n";
```

Comment:
I suggest adding a legend to help with understanding:
```suggestion
    std::string ret = "E: Extra exercise\nA: Active exercise\nM: Moderate exercise\nL: Light exercise\nN: No exercise\n\n      gained     spent      total\n";
```

Reply:
Done, and made some other adjustments to make it fixed width.
![image](https://user-images.githubusercontent.com/42699974/87857793-55713980-c8de-11ea-9d0d-fbafe83f9af6.png)